vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/cache/CacheControl.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2021 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.web.client.impl.cache;
+
+import io.netty.handler.codec.DateFormatter;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpHeaders;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Parse HTTP headers to determine if and how to cache a response.
+ *
+ * @author <a href="mailto:craigday3@gmail.com">Craig Day</a>
+ */
+public class CacheControl {
+
+  private static final String SHARED_MAX_AGE = "s-maxage";
+  private static final String MAX_AGE = "max-age";
+
+  private final Set<String> directives;
+  private final Map<String, Long> timeDirectives;
+  private final Date expires;
```

Comment:
can we use the new immutable date / time API instead of java.util.Date ?

Reply:
yeah, I think so. I did this because the netty class that can parse/encode http date headers uses java.util.Date, but I think it will be straightforward enough to convert to/from them.

Reply:
thanks

Reply:
Updated this to use `Instant` and `Duration`, please un-resolve this comment if something else is desired.